### DISCUSSION TITLE
Does Go Micro need a website?

### DISCUSSION THREAD

**Discussion by asim** [micro/go-micro#2421] [Ideas]
Started to wonder whether we need a website that has  examples/docs/plugins/tutorials, etc. Next.js is a very popular frontend framework which as a nice website. Any thoughts?

**Comment by refs** [comment]
Uuuh, a website promoting the examples the way NextJS looks very inviting for developers. Having such examples from "just an example" to actually having something deployed is very very fast the way they have it set up; I'm struggling to find of an equivalent with go-micro. I just wonder if having examples very accessible and readily deploy-able would "alienate" developers from the fact that there are multiple micro-services under the hoods.

On the other hand having examples advertising the most solid features and the go-micro way of building services would be a more appropriate content for users of the library. The organization could look somehow similar to that of the nextjs website, but the examples can actuaslly follow a progression. A progression such as:

1. Create a very basic go-micro service
2. Create a go-micro client
  2.1 Send a message to the service using a go-micro client from another module
3. Configure the go-micro client
4. Adding a gateway
  4.1 Using a gateway to send messages to services
...

And so on. It will require a good deal of organization, but IMHO it aids with the learning curve and discovery. If one wishes to dig deeper, there is also each individual section well documented with code and READMEs. My 2️⃣  cents! would love to add more!

**Comment by hhstore** [comment]
>  About the documentation. 

- `vitepress` is recommended, supports multiple languages. this is the official project of `vue.js`.
    - https://github.com/vuejs/vitepress
    - https://github.com/vuejs/vuepress
- https://github.com/hhstore/blog/issues/119
- https://github.com/hhstore/blog/issues/332
- Go-micro's past documentation, too confusing. (Because of the relationship between go-micro and micro, and the complexity itself). 

> All-In-One Use Case.

- My personal opinion, go-micro's complexity and flexibility, requires an `all-in-one` product example project. Include the usage of the main features. 
- Such as an e-commerce platform. Or a small forum project.  Other similar frameworks ([go-kratos](https://github.com/go-kratos/kratos), [go-zero](https://github.com/zeromicro/go-zero)) are doing this. 
    - https://github.com/go-kratos/beer-shop
    - https://github.com/zeromicro/zeromall
- Ordinary examples, for use cases for a single function point, still can't explain a complex and flexible large project, like `go-micro`.
- Frankly, for junior engineers, `go-micro` is very difficult to get started with.
